library/unwind/build.rs

```diff
@@ -57,101 +58,119 @@ mod llvm_libunwind {
+                }
+            }
+            cpp_cfg.flag("-nostdinc++");
+            cpp_cfg.flag("-fno-exceptions");
+            cpp_cfg.flag("-fno-rtti");
+            cpp_cfg.flag_if_supported("-fvisibility-global-new-delete-hidden");
+
+            for cfg in [&mut cc_cfg, &mut cpp_cfg].iter_mut() {
+                cfg.flag("-fstrict-aliasing");
+                cfg.flag("-funwind-tables");
+                cfg.flag("-fvisibility=hidden");
+                cfg.define("_LIBUNWIND_DISABLE_VISIBILITY_ANNOTATIONS", None);
+            }
         }
 
-        let mut unwind_sources = vec![
-            "Unwind-EHABI.cpp",
-            "Unwind-seh.cpp",
+        let mut c_sources = vec![
             "Unwind-sjlj.c",
             "UnwindLevel1-gcc-ext.c",
             "UnwindLevel1.c",
             "UnwindRegistersRestore.S",
             "UnwindRegistersSave.S",
-            "libunwind.cpp",
         ];
 
-        if target_vendor == "apple" {
-            unwind_sources.push("Unwind_AppleExtras.cpp");
-        }
+        let cpp_sources = vec!["Unwind-EHABI.cpp", "Unwind-seh.cpp", "libunwind.cpp"];
+        let cpp_len = cpp_sources.len();
 
         if target.contains("x86_64-fortanix-unknown-sgx") {
-            unwind_sources.push("UnwindRustSgx.c");
+            c_sources.push("UnwindRustSgx.c");
         }
 
         let root = Path::new("../../src/llvm-project/libunwind");
-        cfg.include(root.join("include"));
-        for src in unwind_sources {
-            cfg.file(root.join("src").join(src));
+        cc_cfg.include(root.join("include"));
+        cpp_cfg.include(root.join("include"));
+        for src in c_sources {
+            cc_cfg.file(root.join("src").join(src));
         }
 
-        if target_env == "musl" {
-            // use the same C compiler command to compile C++ code so we do not need to setup the
-            // C++ compiler env variables on the builders
-            cfg.cpp(false);
-            // linking for musl is handled in lib.rs
-            cfg.cargo_metadata(false);
-            println!("cargo:rustc-link-search=native={}", env::var("OUT_DIR").unwrap());
+        for src in cpp_sources {
+            cpp_cfg.file(root.join("src").join(src));
         }
 
-        cfg.compile("unwind");
+        cpp_cfg.cargo_metadata(false);
+        cc_cfg.cargo_metadata(false);
+        let out_dir = env::var("OUT_DIR").unwrap();
+        println!("cargo:rustc-link-search=native={}", &out_dir);
+
+        cpp_cfg.compile("unwind-cpp");
+
+        let ar = env::var("AR")
+            .or(env::var(format!("AR_{}", target)))
+            .unwrap_or_else(|_| "ar".to_owned());
+        let obj_dir = Path::new(&out_dir).join("obj");
+        std::fs::create_dir_all(&obj_dir).unwrap();
+        let mut cmd = std::process::Command::new(ar)
+            .args(&["x", &format!("{}/libunwind-cpp.a", &out_dir)])
+            .current_dir(&obj_dir)
+            .spawn()
+            .expect("ar command failed to run");
```

Comment:
Sigh, I was going to suggest compiling into object files (`gcc -c`) instead of a static library to avoid unarchiving, but looks like the `cc` crate doesn't support even that.